src/app/components/common/topbar/topbar.component.ts

```diff
@@ -540,4 +560,113 @@ export class TopbarComponent extends ViewControllerComponent implements OnInit,
   openIX() {
     window.open('https://www.ixsystems.com/', '_blank')
   }
+
+  showTCStatus() {
+    this.tcConnected ? this.openStatusDialog() : this.openSignupDialog();
+  }
+
+  openSignupDialog() {
+    const conf: DialogFormConfiguration = {
+      title: helptext.signupDialog.title,
+      fieldConfig: [
+        {
+          type: 'paragraph',
+          name: 'message',
+          paraText: helptext.signupDialog.content
+        }
+      ],
+      saveButtonText: helptext.signupDialog.connect_btn,
+      custActions: [
+        {
+          id: 'signup',
+          name: helptext.signupDialog.singup_btn,
+          function: () => {
+            window.open('https://portal.ixsystems.com');
+            this.dialogService.closeAllDialogs();
+          }
+        }
+      ],
+      parent: this,
+      customSubmit: function (entityDialog) {
+        entityDialog.dialogRef.close();
+        entityDialog.parent.updateTC();
+      }
+    }
+    this.dialogService.dialogForm(conf);
+  }
+
+  updateTC() {
+    const self = this;
+    const conf: DialogFormConfiguration = {
+      title: self.tcConnected ? helptext.updateDialog.title_update : helptext.updateDialog.title_connect,
+      fieldConfig: [
+        {
+          type: 'input',
+          name: 'api_key',
+          placeholder: helptext.updateDialog.api_placeholder,
+          tooltip: helptext.updateDialog.api_tooltip,
+        },
+        {
+          type: 'checkbox',
+          name: 'enabled',
+          placeholder: helptext.updateDialog.enabled_placeholder,
+          tooltip: helptext.updateDialog.enabled_tooltip,
+          value: true,
+        }
+      ],
+      saveButtonText: self.tcConnected ? helptext.updateDialog.save_btn : helptext.updateDialog.connect_btn,
```

Comment:
I am not sure how this shows up in UI, but if the user changes the api key - then essentially we are starting the process afresh by registering the key with the portal and then polling. So I think it might be nice if we handle it like that